### DISCUSSION TITLE
Как на вход хранимой процедуре дать XML-ный файл?

### DISCUSSION THREAD

**Discussion by tsv19su254052** [tsv19su254052/LoadWorkData-GUIs-and-Utilities#109] [Show and tell]
Как в графической оболочке или программно дать на вход хранимой процедуре XML-ный файл например через диалог открытия файла из старой XML-ной базы данных?

![SSMS Даем на вход хранимой процедуре XML-ный файл](https://user-images.githubusercontent.com/104857185/175548407-cc62fdc3-bb65-432a-b8f7-b4a941fffd28.png)

_Originally posted by @tsv19su254052 in https://github.com/tsv19su254052/LoadWorkDataAirFlightsDBNew/issues/1#issuecomment-1165589900_

**Comment by jvxram** [comment]
Не хотелось бы персонально для ввода первички делать графическую оболочку

**Comment by tsv19su** [comment]
Исходник старой XML-ной базы летательного аппарата может быть **размером до 2 Гбайт** и поэтому может быть проблематично вставить его через буфер обмена, как первичку

**Comment by tsv19su254052** [comment]
А SSMS примет 5 * 2 Гбайт?
Было ограничение: на все про все  - 2 Гбайта

**Comment by jvxram** [comment]
Надо почитать статьи по приведенным ранее ссылкам про XPath и EXPath, а также про EXSLT

**Reply by tsv19su254052** [reply]
Почитаю

**Comment by tsv19su254052** [comment]
Даю пути до исходных файлов. Просто вставляю их через буфер обмена, так как диалога открытия файла (с фильтром например *.xml) в графической формочке хранимой процедуры нет

![SSMS Даем на вход хранимой процедуре XML-ный файл 002](https://user-images.githubusercontent.com/104857185/217669151-61f4816e-8e7b-461f-86eb-2f992fe73bb3.png)

И выдает такие ошибки

![SSMS Даем на вход хранимой процедуре XML-ный файл 003](https://user-images.githubusercontent.com/104857185/217669418-663c44a1-bb8c-40e3-b355-8e29feb6f30a.png)
